src/goDocumentation.ts

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+import vscode = require('vscode');
+import cp = require('child_process');
+import path = require('path');
+import { CancellationToken, TextDocumentContentProvider, Uri } from 'vscode';
+import { getBinPath, getToolsEnvVars, byteOffsetAt, getFileArchive } from './util';
+
+const missingToolMsg = 'Missing tool: ';
+
+export function getDocumentation(): void {
+	let gogetdoc = getBinPath('gogetdoc');
+	if (!path.isAbsolute(gogetdoc)) {
+		vscode.window.showErrorMessage(missingToolMsg + 'gogetdoc');
+		return;
+	}
+
+	let ate = vscode.window.activeTextEditor;
+	if (typeof ate !== 'undefined') {
+		let document = ate.document;
+		let gogetDocsFlags = ['-u', '-json', '-modified', '-pos', document.fileName + ':#' + byteOffsetAt(document, ate.selection.active).toString()];
+		let p = cp.execFile(gogetdoc, gogetDocsFlags, {env: getToolsEnvVars()}, (err, stdout, stderr) => {
+			try {
+				let goGetDocOutput = <GoGetDocOutput>JSON.parse(stdout.toString());
+
+				// drop the stuff before /vendor/ if we have it
+				let imprt = goGetDocOutput.import.replace(/(.*\/vendor\/)?(.*)/, '$2');
+
+				_getDocumentationForImport(imprt);
+			} catch (e) {
+				_getDocumentationForImport('');
+			}
+		});
+		p.stdin.end(getFileArchive(document));
+	} else {
+		_getDocumentationForImport('');
+	}
+}
+
+function _getDocumentationForImport(imprt: string): void {
+	vscode.window.showInputBox({
+		prompt: 'Please enter a package name',
+		value: imprt,
+		placeHolder: imprt === '' ? 'no package detected' : '',
+	}).then(pkgInput => {
+		if (typeof pkgInput === 'undefined' || pkgInput === '') {
+			return;
+		}
+
+		let uri = vscode.Uri.parse('godocumentation://');
+		uri = uri.with({
+			path: pkgInput,
+		});
+
+		vscode.workspace.openTextDocument(
```

Comment:
you can directly call `showTextDocument` here

`vscode.window.showTextDocument(uri, { viewColumn: vscode.window.activeTextEditor.viewColumn + 1, preserveFocus: true });`